### PR TITLE
Delete pod from pod controllers route.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This implements the removal of pod through the pods controller.

JIRA issues: MARATHON-7877

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala [hunk 1]
```diff
@@ -100,7 +100,36 @@ class PodsController(
 
   def find(podId: PathId): Route = ???
 
-  def remove(podId: PathId): Route = ???
+  def remove(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      (extractClientIP & forceParameter) { (clientIp, force) =>
+        extractRequest { req =>
```
Comment (meln1k) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: Can we extract URI instead of request here?

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: Sure. Much better. Thanks.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala [hunk 2]
```diff
@@ -100,7 +100,37 @@ class PodsController(
 
   def find(podId: PathId): Route = ???
 
-  def remove(podId: PathId): Route = ???
+  @SuppressWarnings(Array("all")) // async/await
+  def remove(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      (extractClientIP & forceParameter) { (clientIp, force) =>
```
Comment (meln1k) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: What do you think about this?
```scala
(extractClientIP & forceParameter & extractUri) { (clientIp, force, uri) =>
```

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala [hunk 3]
```diff
@@ -55,7 +59,12 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
       val request = Post(Uri./.withQuery(Query("force" -> "false")))
         .withEntity(entity)
         .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      val podDefinition = PodDefinition(id = PathId("mypod"))
+      f.podManager.find(any).returns(Some(podDefinition))
```
Comment (meln1k) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: How about this? `f.podManager.find(any)` -> `f.podManager.find(eq(PathId("mypod")))`

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala (hunk 2 above)
Comment (alenkacz) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: +1

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: ✅

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala (hunk 3 above)
Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: Same as in #5754. We do not want to test whether `find` was invoked but if the request is rejection due to unauthorized requests.

**Pull request merged**